pkgs/by-name/co/cook-cli/package.nix

```
@@ -0,0 +1,54 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  buildNpmPackage,
+  rustPlatform,
+  pkg-config,
+  openssl,
+  darwin,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "cook-cli";
+  version = "0.7.0";
+  src = source;
+  cargoHash = "sha256-YOmEGnSOG8uG0fLVTxF6hDKp1sPXoCuSuPRMLoJU1QM=";
+  nativeBuildInputs = [pkg-config openssl];
+  buildInputs =
+    [openssl]
+    ++ lib.optionals stdenv.isDarwin [darwin.apple_sdk.frameworks.SystemConfiguration];
+  postPatch = ''
+    cp -r ${ui}/* "ui/public"
+  '';
+  OPENSSL_NO_VENDOR = 1;
+
+  source = fetchFromGitHub {
+    owner = "cooklang";
+    repo = "cookcli";
+    rev = "v${version}";
+    hash = "sha256-dPmJ+BMNZfzzLqA4dYvf47Qb6jsgnW67qqjvSWLkyPs=";
+  };
+
+  ui = buildNpmPackage {
+    name = "ui";
+    src = "${src}/ui";
+    npmDepsHash = "sha256-OTSes4yratfPJippeKfXQSyRmdYw0MbGHU42LzVugl4=";
+    makeCacheWritable = true;
+    npmFlags = ["--legacy-peer-deps"];
+    installPhase = ''
+      runHook preInstall
+      mv public/ $out
+      runHook postInstall
+    '';
+  };
+
+  meta = with lib; {
+    changelog = "https://github.com/cooklang/cookcli/releases/tag/v${version}";
+    description = "Command line program which provides a suite of tools to create shopping lists and maintain recipes.";
```

Since you're providing a GUI as well:
```suggestion
    description = "A suite of tools to create shopping lists and maintain recipes";
```